src/grpc/infer_handler.h

```diff
@@ -669,6 +670,33 @@ class InferHandlerState {
       return received_notification_ ? ctx_->IsCancelled() : false;
     }
 
+    // Extracts headers from GRPC request and updates state
+    void ExtractStateFromHeaders(InferHandlerStateType* state)
+    {
+      const auto& metadata = state->context_->ctx_->client_metadata();
+      for (const auto& pair : metadata) {
+        auto& key = pair.first;
+        std::string param_key = std::string(key.begin(), key.end());
+        std::string grpc_strict_key = "grpc_strict";
+        if (param_key.compare(grpc_strict_key) == 0) {
+          state->context_->grpc_strict_ = true;
+        }
+      }
+    }
+
+    void sendGRPCStrictResponse(InferHandlerStateType* state)
+    {
+      std::lock_guard<std::recursive_mutex> lock(state->context_->mu_);
+      // Check if Error not responded previously
+      // Avoid closing connection twice on multiple errors from core
+      if (!state->context_->IsGRPCStrictError()) {
+        state->context_->responder_->Finish(state->status_, state);
+        // Mark error for this stream
+        state->context_->MarkGRPCStrictError();
+        // Fix Me : Last argument not sure for HandleCancellation
```

Comment:
Still a FIXME?

Reply:
Yes HandleCancellation needs a string name_, which it uses for logging.
As I have added this inside stream_infer_handler.cc which does not have a Name() function, hence "grpc_strict_name" as a placeholder